common/remote/sockfile.cpp

```diff
@@ -2719,42 +2732,77 @@ ISocket *connectDafs(SocketEndpoint &ep, unsigned timeoutms)
     if ( (securitySettings.connectMethod == SSLNone) || (securitySettings.connectMethod == SSLOnly) )
     {
         socket.setown(ISocket::connect_timeout(ep, timeoutms));
-        return socket.getClear();
+        return checkSocketSecure(socket);
     }
 
+    // SSLFirst or UnsecureFirst ...
+
     unsigned newtimeout = timeoutms;
     if (newtimeout > 5000)
         newtimeout = 5000;
 
-    bool tryAgain = false;
-    try
+    int conAttempts = 2;
+    while (conAttempts > 0)
     {
-        socket.setown(ISocket::connect_timeout(ep, newtimeout));
-    }
-    catch (IJSOCK_Exception *e)
-    {
-        if (e->errorCode() == JSOCKERR_connection_failed)
+        conAttempts--;
+        bool connected = false;
+        try
         {
-            unsigned prevPort = ep.port;
-            if (prevPort == securitySettings.daFileSrvSSLPort)
-                ep.port = securitySettings.daFileSrvPort;
+            socket.setown(ISocket::connect_timeout(ep, newtimeout));
+            connected = true;
+            newtimeout = timeoutms;
+        }
+        catch (IJSOCK_Exception *e)
+        {
+            if (e->errorCode() == JSOCKERR_connection_failed)
+            {
+                e->Release();
+                if (ep.port == securitySettings.daFileSrvSSLPort)
+                    ep.port = securitySettings.daFileSrvPort;
+                else
+                    ep.port = securitySettings.daFileSrvSSLPort;
+                if (!conAttempts)
+                    throw;
+            }
             else
-                ep.port = securitySettings.daFileSrvSSLPort;
-            WARNLOG("Connect failed on port %d, retrying on port %d", prevPort, ep.port);
-            tryAgain = true;
-            e->Release();
+                throw;
+        }
+
+        if (connected)
+        {
+            if (ep.port == securitySettings.daFileSrvSSLPort)
+            {
+                try
+                {
+                    return checkSocketSecure(socket);
+                }
+                catch (IDAFS_Exception *e)
+                {
+                    connected = false;
+                    if (e->errorCode() == DAFSERR_connection_failed)
+                    {
+                        // worth logging to help identify any ssl config issues ...
+                        StringBuffer errmsg;
+                        e->errorMessage(errmsg);
+                        WARNLOG("%s", errmsg.str());
+                        e->Release();
+                        ep.port = securitySettings.daFileSrvPort;
```

Comment:
connectDafs() needs to update the ep.port so that spray() / despray() that happen later on have the correct ep.